### PR TITLE
Fixing scale up issue in service with custom vnets

### LINKED ISSUE #1195: K8s custom vnet scale up is broken
**Is this a request for help?**:
No
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE
---

**What version of acs-engine?**:
current
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
k8s

There are several places in the template that refers to the NSG. In case of cluster deployed in a custom VNET the references to the NSG differ from the case of default VNET template. The template transformer was written for the case of default VNET only and needs to be updated to account for the case when a custom VNET is specified and remove those dependencies correctly during scale up.

### PR DESCRIPTION
**Pull request by JackQuincy**
**What this PR does / why we need it**:
Fixes a bug
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
fixes #1195
**Special notes for your reviewer**:

**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1194)

### CODE REVIEW

File: examples/scale-up/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ outages in any services running on the agent pool and/or fail the update deploym
 To get around this there is now an Offset parameter on each agent pool. If you want to scale up an existing cluster take the template 
 you deployed orignally. Update the accompanying parameters file to have the Offset == old count and the Count == new desired number of vms. 
 example: I originally deployed 5 vms in an agent pool. I want to scale up to 10. I would set Offset==5 and Count == 10. Then deploy the template 
-in incremental mode. This sadly isn't every change that is needed. We also need to remove the NSG resource and the vnet's dependency on it. The current design of ACS-engine doesn't allow for a seperate update template, so this will have to be a manual step. I have a sample update template here.
+in incremental mode. This sadly isn't every change that is needed. We also need to remove the NSG resource and the vnet's dependency on it. If this is a custom VNET/bring your own vnet case, you need to remove the dependecies from each of the agent pools and the vm master loop. The current design of ACS-engine doesn't allow for a seperate update template, so this will have to be a manual step. I have a sample update template here.
```
Comment (amanohar) [inline examples/scale-up/README.md]: Fix spelling

Comment (seanknox) [inline examples/scale-up/README.md]: Please remove the "This sadly isn't every change that is needed."

File: pkg/acsengine/transform.go [hunk 2]
```diff
@@ -111,7 +114,7 @@ func (t *Transformer) NormalizeForK8sVMASScalingUp(logger *logrus.Entry, templat
 
 		for dIndex := len(dependencies) - 1; dIndex >= 0; dIndex-- {
 			dependency := dependencies[dIndex].(string)
-			if strings.Contains(dependency, nsgResourceType) {
+			if strings.Contains(dependency, nsgResourceType) || strings.Contains(dependency, nsgID) {
```
Comment (seanknox) [inline pkg/acsengine/transform.go]: We need a unit test with this fix.

Comment (JackQuincy) [inline pkg/acsengine/transform.go]: I added one. It took a lot of work as when this code was moved from the service code to acs-engine the unit tests weren't fully brought over.

**Pull request merged**